app/components/graph-editor/math.ts

```diff
@@ -0,0 +1,29 @@
+// File: math.ts
+// Created by: CJ Dimaano
+// Date created: January 9, 2016
+//
+// THIS FILE IS INTENDED TO BE IMPORTED ONLY INTO graph-editor.component.ts
+
+
+/**
+ * dot  
+ *   Calculates the dot product of two vectors.
+ */
+export function dot(a: number[], b: number[]): number {
```

Comment:
This seems like it could be useful for the ML plugin if we decide to have an option for running as JS. Also, this seems like potentially a good place to bring in a library for linear algebra stuff if only for performance. Something like ![vectorius](https://github.com/mateogianolio/vectorious).